agent/connect/x509_patch.go

```diff
@@ -0,0 +1,122 @@
+package connect
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/url"
+	"unicode"
+)
+
+// NOTE: the contents of this file were lifted from
+// $GOROOT/src/crypto/x509/x509.go from a Go 1.16.5 checkout.
+//
+//
+// After https://go-review.googlesource.com/c/go/+/329129 lands in a Go release
+// we are compiling against we can safely remove all of this code.
+
+var (
+	x509_oidExtensionSubjectAltName = []int{2, 5, 29, 17}
+)
+
+const (
+	x509_nameTypeEmail = 1
+	x509_nameTypeDNS   = 2
+	x509_nameTypeURI   = 6
+	x509_nameTypeIP    = 7
+)
+
+func HackSANExtensionForCSR(template *x509.CertificateRequest) {
```

Comment:
This is basically attempting to repeat this blob of code from the stdlib ourselves:
https://github.com/golang/go/blob/0e67ce3d28320e816dd8e7cf7d701c1804fb977e/src/crypto/x509/x509.go#L1088

Reply:
Should there be a comment for this function that includes the link as well?

Reply:
:+1: